martini_sour/src/analyze.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2021 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import MDAnalysis as mda
+from tqdm import tqdm
+import matplotlib.pyplot as plt
+from scipy.optimize import curve_fit
+from .titrate import get_ph_values
+
+
+def setup_dirs(ph_range, sim_dir, traj_file, tpr_file):
```

Comment:
I've some issues with this function. First I think for path-handling here and everywhere else, I recommend using pathlib or the os internal path library, because using strings in place of paths will make the program dependent on the OS. It makes our life much easier to use proper Paths objects.

Second, not all users will give standard names to their titration directories. It would be more fool-proof to say loop over directories inside a directory - have a look at os.walk (see https://docs.python.org/3/library/os.html) - and extract the pH values either from the top-file or the directory name. Then setting the pH increment and values becomes independent of the analysis script. In future we might want to extend the range and or skip values. Perhaps some values fail.